astropy/units/format/base.py

```diff
@@ -39,6 +43,8 @@ def __init_subclass__(cls, **kwargs):
         # Keep a registry of all formats.  Key by the class name unless a name
         # is explicitly set (i.e., one *not* inherited from a superclass).
         if "name" not in cls.__dict__:
+            if cls.__name__.endswith("Base"):
```

Comment:
I agree this is janky. I think the way to solve this is instead to check for a class being abstract or not — `inspect.isabstract`. This would of course mean that we make all the base classes (BCs) into ABCs. But I think we should do that!

Reply:
What abstract classes are supposed to mean is that they have abstract methods the presence of which prevents creating instances directly, but it is possible to instantiate subclasses of abstract classes if they have implementations for all the abstract methods. The `astropy` unit formatters are different because they are never meant to be instantiated, no matter what they implement. Furthermore, the implementation of `Base.__new__()` would conflict with `ABC.__new__()`.

Luckily for us, there already exists a naming convention for things that should not be public:
```suggestion
            if cls.__name__.startswith("_"):
```